### DISCUSSION TITLE
Design changes needed to support local plan data

### DISCUSSION THREAD

**Discussion by psd** [digital-land/digital-land#349] [Ideas and feedback]
This discussion sketches out design changes needed to support browsing and searching for local plans on https://planning.data.gov.uk

# Context
We have begun moving data from the [local plans prototype](https://local-plans-prototype.herokuapp.com/local-plans) onto https://planning.data.gov.uk

This means local plans content on GOV.UK will to be able to link to https://planning.data.gov.uk along the following lines:

> Use the [planning data platform](https://submit.planning.data.gov.uk/guidance/specifications/local-plan) as a guide to provide data about your plan. 

This is a link to guidance on how an LPA can provide and update the data  on their website, which links to the part of the planning data platform which helps them through the process, and gives feedback on any with their data.

> Providing data about your plan makes it available on the [national map of local plan areas](https://www.planning.data.gov.uk/map/?dataset=local-plan-boundary) and the data findable in the [national list of local plans](https://www.planning.data.gov.uk/entity/?dataset=local-plan).

That is separate links to the map and search pages on planning.data.gov.uk:

![Local plan navigation](https://github.com/user-attachments/assets/66394f75-c05d-49b1-95e3-2f6607681415)

**Comment by psd** [comment]
# Data design process
A data model is currently being tested using data imported from the local plans prototype and reviewed using the from existing local plans using a new [development plans data prototype](https://local-plans-explorer-0d7b65a24242.herokuapp.com/).

A specification for how LPAs can provide this data is under development using our open co-design process:
* [specification](https://digital-land.github.io/specification/specification/local-plan/) for LPAs to provide information about their local plan including links to documents and its timetable currently set out in the Local Development Scheme.
* [consideration](https://design.planning.data.gov.uk/planning-consideration/development-plans-and-timetables) 
* [discussion](https://github.com/digital-land/data-standards-backlog/discussions/26) 

Other datasets relating to recording policies, site allocations, and metrics such as the target housing number are in development: https://design.planning.data.gov.uk/planning-consideration/

# Data model

![Data model](https://digital-land.github.io/specification/specification/local-plan/diagram.svg)

Data collected during the data design process and provided by LPAs is made available on planning.data.gov.uk in a data model consisting of four datasets:

* [local-plan-boundary](https://www.planning.data.gov.uk/dataset/local-plan-boundary)—the boundary is often the local planning authority boundary, though some plans may have different boundaries when published jointly for example. There may one or many plans for a single local plan area. 
* [local-plan](https://www.planning.data.gov.uk/dataset/local-plan)—the name and date range for the local plan with a link to the local-plan-boundary
* [local-plan-document](https://www.planning.data.gov.uk/dataset/local-plan-document)—core and supplementary strategy documents, notices and other documents linked to the plan.
* [local-plan-timetable](https://www.planning.data.gov.uk/dataset/local-plan-timetable)—planned and actual dates of events such as when a plan is submitted, adopted or withdrawn which make up the timetable of producing the local plan

**Reply by AdeleMplan** [reply]
From a user perspective, local plans timetable implies a thing like an LDS i.e. my timetable is to be at stage A by this date, stage B by this date, stage C by this date. Submitted, adopted, withdrawn read more like a status. If you know what i mean.

**Reply by AdeleMplan** [reply]
Great to see all this work by the way. Exciting.

**Comment by psd** [comment]
# The map page
The [map page](https://www.planning.data.gov.uk/map/?dataset=local-plan-boundary&dataset=local-planning-authority#52.367744489723776,0.32278405562124135,7.9614560418241345z) lets you browse local plan boundaries, including joint plans:

<kbd>![Screenshot From 2024-11-15 14-11-06](https://github.com/user-attachments/assets/a93443ee-1b78-4881-bbc9-a14cbb52b15c)</kbd>

This dataset will comprise, many of which are the LPA boundary, with some exceptions, such as joint local plans. Clicking on the map presents a link to the local plan boundary entity.

**Comment by psd** [comment]
# Local plan boundary entity page

Each local plan boundary has an entity page, for example [South Cambridgeshire](https://www.planning.data.gov.uk/entity/4211006) for a local-plan-boundary. 

This is currently a generic entity page, highlighting the data with a map of the boundary.

There may be one or more local plans which cite an area as their boundary. These entities link to the boundary, so currently aren't reachable from the local plans boundary entity.

The boundary page template needs a table of links to local plans adding.

We could also look at a line of content at the top of the page, giving a summary of the data.

**Reply by Ben-Hodgkiss** [reply]
[Issue raised](https://github.com/digital-land/technical-documentation/issues/153) for Infrastructure Team to work on in upcoming sprint.

**Comment by psd** [comment]
# Local plan entity page

Each local plan has an entity page, for example [South Cambridgeshire](https://www.planning.data.gov.uk/entity/4211006).

This is currently a generic entity page with some basic data about the plan, its name, responsible organisations.

The `documentation-url` field links to the local plan on the LPA website.

The `local-plan-boundary` field links to the boundary entity. The template be changed to also show the boundary as a map.

The template needs a table of `local-plan-document` entities which link the the local plan. This table can link to the document entities and include the name, category and direct link to the documents on the LPA website. 

Similarly the template should display `local-plan-timetable` entities which link to this plan. This will initially be a table, with events in date order, and can be improved upon.

We could also look at a line of content at the top of the page, giving a summary of the data.

**Reply by eveleighoj** [reply]
When you say:

The template be changed to also show the boundary as a map.

do you want to also show the local-plan-boundary on the local-plan page or just on the local-plan-boundary page?

**Reply by psd** [reply]
On both. The boundary is already visible on the local-plan-boundary page. We can also show the boundary for entities where the `dataset` is `local-plan`.

**Reply by Ben-Hodgkiss** [reply]
[Issue raised](https://github.com/digital-land/technical-documentation/issues/154) for Infrastructure Team to work on in next sprint.

**Comment by psd** [comment]
# The list of local plans

The current [entity search page](https://www.planning.data.gov.uk/entity/)/

The page will be enhanced to allow faceting by the responsible organisation.

The design of the list result box can be improved.

**Reply by psd** [reply]
We can also look at how to filter by the status of the local plan, but that needs a little work.

**Reply by Ben-Hodgkiss** [reply]
[Initial ticket](https://github.com/digital-land/technical-documentation/issues/155) created here for Infrastructure to look at in the next sprint. If scope expands for further filtering (e.g. by status) then let me know.

**Comment by psd** [comment]
# Site as of January 2025

The map has a selection of boundaries where they match the LPA boundary:
<kbd>![image](https://github.com/user-attachments/assets/7a7d12eb-d458-4bde-906e-0994a47c347a)</kbd>

The boundary entity has a link to plans in the area, eg [Bassetlaw]https://www.planning.data.gov.uk/entity/4211019):
<kbd>![image](https://github.com/user-attachments/assets/30078a59-5387-4dd0-a860-4abd06466e83)</kbd>

And the [local plan entity](https://www.planning.data.gov.uk/entity/4220020) has the map of the boundary, links to the plans, their timetable, and links to the documents:
<kbd>![image](https://github.com/user-attachments/assets/bb9df6ed-f3b2-4949-b5cb-6170c4a98a5c)</kbd>
